lib/manageiq/api/client/mixins/action_mixin.rb

```diff
@@ -2,6 +2,16 @@ module ActionMixin
   extend ActiveSupport::Concern
 
   private
+  def find_action(name)
+    @action = @actions.find { |action| action.name == name.to_s }
+  end
+
+  def do_action(name)
+    @action ||= find_action(name)
+    client.send @action.method, URI(@action.href) do
+      { action: @action.name }
```

Comment:
needs to support

```ruby
{
     :action => action_name,
     :resource => {
        param1 => value1,
        param2 => value2,
        ...
     }
}
```

param1, param2, ... if any options are passed to the action (methd_missing args, &block).